vcd/resource_vcd_catalog.go

```diff
@@ -0,0 +1,132 @@
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/govcd"
+	"log"
+)
+
+func resourceVcdCatalog() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdCatalogCreate,
+		Delete: resourceVcdCatalogDelete,
+		Read:   resourceVcdCatalogRead,
+		Update: resourceVcdCatalogUpdate,
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vdc": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				ForceNew: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				ForceNew: true,
+			},
+			"delete_force": &schema.Schema{
+				Type:        schema.TypeBool,
+				Required:    true,
+				ForceNew:    false,
+				Description: "force=True along with recursive=True to remove a catalog and any objects it contains, regardless of their state.",
+			},
+			"delete_recursive": &schema.Schema{
+				Type:        schema.TypeBool,
+				Required:    true,
+				ForceNew:    false,
+				Description: "recursive=True to remove a Catalog and any objects it contains that are in a state that normally allows removal.",
```

Comment:
Please adjust description to mention that this is for `destroy` operation as above.

BTW, which state do not "normally allow removal"? And will anything be deleted if there are such items in a catalog?

Reply:
Regarding BTW: normally any state do not allow removal. Recursive means delete catalog even items exists. Force allows to delete if item in not success state. Not exact definition in api. I tested with just failed upload.

Reply:
Description updated.